File: sketch/wc.py
#!/usr/bin/env python3

import sys
import io

###################################################################################################
def wc_by_line(src : io.TextIOWrapper):
    line_count = 0

    for _ in src:
        line_count += 1

    print(line_count, end=" ")

    if line_count == 1:
        print("line")
    else:
        print("lines")




###################################################################################################
def main(argc, argv):
    if argc > 1:
        for fname in argv[1:]:
            try:
                src = open(fname, 'r')
                wc_by_line(src)
                src.close()
            except FileNotFoundError:
                pass
    else:
        wc_by_line(sys.stdin)

    return 0



###################################################################################################
if __name__ == "__main__":
    sys.exit(main(len(sys.argv), sys.argv))

File: cpp/sketch/wc.cpp
#include <iostream>
#include <fstream>

/*************************************************************************************************/
void wc_by_line(std::istream& src) {
    std::string line;
    int line_count = 0;

    while (std::getline(src, line)) {
        line_count ++;
    }

    std::cout << line_count;

    if (line_count == 1) {
        std::cout << " line";
    } else {
        std::cout << " lines";
    }

    std::cout << std::endl;
}



/*************************************************************************************************/
int main(int argc, char* argv[]) {
    if (argc > 1) {
        for (int idx = 1; idx < argc; idx++) {
            std::ifstream src(argv[idx]);

            if (src.is_open()) {
                wc_by_line(src);
                src.close();
            }
        }
    } else {
        std::cout << "[read from user input]" << std::endl;
        wc_by_line(std::cin);
    }

    return 0;
}
